### DISCUSSION TITLE
How to use multisampling

### DISCUSSION THREAD

**Discussion by coldspice** [gfx-rs/wgpu#4860] [General]
when i set sampling count to 4 in render Pipeline i get this error

  ```
  Render pipeline targets are incompatible with render pass
    Incompatible sample count: the RenderPass uses textures with sample count 1 but the RenderPipeline uses attachments with format 4

```
i'm doing 
```

let  mut output = self.device.surface.get_current_texture()?;
let view=output.texture.create_view(&wgpu::TextureViewDescriptor::default());
...
draw
...
output.present();
```
self.device.surface.get_current_texture() always return surfaceTexture with sample_cout=1 how do i change it to 4

**Comment by lisyarus** [comment]
You need to manually create a texture with the same format as the surface texture, but with 4 samples. Then, when creating a render pass, you set the color attachment view to your multisampled texture, and the resolve target to the surface texture.
